certora/specs/SocialRecoveryModule.spec

```
@@ -81,6 +79,36 @@ function requireGuardiansLinkedListIntegrity(address guardian) {
     require guardianStorageContract.entries[safeContract].count == guardianStorageContract.countGuardians(safeContract);
 }
 
+// Invariant that proves the relationship between the new threshold, new owner length and the
+// `confirmedHash`. If there is a `confirmedHash` for a given `hash` and `guardian`, then the
+// threshold should be greater than zero and less than or equal to the number of new owners.
+invariant approvedHashesHaveCorrectThreshold(address wallet, address[] newOwners, uint256 newThreshold, uint256 nonce, bytes32 hash)
+    hash == getRecoveryHash(wallet, newOwners, newThreshold, nonce) &&
+    ! (forall address guardian. !currentContract.confirmedHashes[hash][guardian]) =>
+    0 < newThreshold && newThreshold <= newOwners.length
```

nit: This condition is hard to read WRT order of operations. Suggestion:
```suggestion
    (!(forall address guardian. !currentContract.confirmedHashes[hash][guardian])) =>
        0 < newThreshold && newThreshold <= newOwners.length
```

---

Also, to make sure I understood:

`(!(forall address guardian. !currentContract.confirmedHashes[hash][guardian]))` is "if there is any confirmed hash" right?

---

Yes, that is my understanding as well, though will confirm before merging.

---

The assumption is correct, but slightly changed it using `exists` for easier comprehension (suggested by the certora team).